ifrs17/Utils/ActivityLog.ipynb

```diff
@@ -354,6 +354,160 @@
             "execution_count": 0,
             "outputs": []
         },
+        {
+            "cell_type": "code",
+            "source": [
+                "public record ImportBuilderWriter(ImportOptionsBuilder Builder, ",
+                "\n                                ISessionVariable Session, ",
+                "\n                                IDataSource DataSource, ",
+                "\n                                IDataSetImportVariable ImportVariable)",
+                "\n{",
+                "\n    public async Task<ActivityLog> ExecuteAsync()",
+                "\n    {",
+                "\n        var log = await Builder.ExecuteAsync();",
+                "\n        var options = Builder.GetImportOptions();",
+                "\n        var activity = new ImportExportActivity(log, Session);",
+                "\n        bool importSucceeded = true;",
+                "\n        switch(options)",
+                "\n        {",
+                "\n            case FileImportOptions fio:",
+                "\n                var importFile = new ImportFile(Guid.NewGuid());",
+                "\n                try",
+                "\n                {",
+                "\n                    importFile = await (new ImportFile(fio, ImportVariable, Session)).InitializeImportDataAsync() as ImportFile;",
+                "\n                    activity = activity with {SourceId = importFile.Id};",
+                "\n                }",
+                "\n                catch (Exception)",
```

Comment:
Shouldn't you also record in the activity the message of the Exception?

Something like

```
catch (Exception ex)
{
   importSucceeded = false;
   activity = activity with {SourceId = null, Exception = ex}; // we can see what we want to save, but at least ex.Message.
}
```